Example/PaymentSheet Example/PaymentSheetUITest/EmbeddedUITest.swift

```diff
@@ -645,10 +645,11 @@ class EmbeddedUITests: PaymentSheetUITestCase {
         cardNumberField.tap()
         app.typeText(XCUIKeyboardKey.delete.rawValue)
         app.buttons["Close"].waitForExistenceAndTap()
-        // ...should de-select the row and clear the payment option
-        XCTAssertFalse(app.staticTexts["Payment method"].waitForExistence(timeout: 1))
-        XCTAssertFalse(app.buttons["New card"].isSelected)
-        XCTAssertFalse(app.buttons["Checkout"].isEnabled)
+        // ...should re-select the last payment option (card)
+        XCTAssertTrue(app.staticTexts["Payment method"].waitForExistence(timeout: 1))
+        XCTAssertEqual(app.staticTexts["Payment method"].label, "•••• 4444")
+        XCTAssertTrue(app.buttons["New card"].isSelected)
+        XCTAssertTrue(app.buttons["Checkout"].isEnabled)
```

Comment:
This is a bug!  We've broken our contract with the merchant - paymentOption is nil now, but we didn't tell them it updated. Their checkout button is enabled, but if they `confirm`, they'll get an assert.

Reply:
Ahh, you're right, good catch, thanks! You're right below, we will need to regenerate the form VC (and maybe even restore the cache to it's previous state?)

I made a change so that if we dismiss the form and it is of the same type we previously had selected we
1. Clear the selecting entirely if the payment option in the form has changed and user hits cancel
2. If the payment option hasn't changed go back to what we have previously selected. We can do this b/c the form matches the payment option and don't need to regenerate the form.